packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -1093,68 +1530,153 @@ class RectangularSliderTrackShape extends SliderTrackShape {
-    // can be a no-op.
-    if (sliderTheme.trackHeight <= 0) {
-      return;
-    }
+    // Assign the track segment paints, which are left: active, right: inactive,
+    // but reversed for right to left text.
+    final ColorTween activeTrackColorTween = ColorTween(begin: sliderTheme.disabledActiveTrackColor , end: sliderTheme.activeTrackColor);
+    final ColorTween inactiveTrackColorTween = ColorTween(begin: sliderTheme.disabledInactiveTrackColor , end: sliderTheme.inactiveTrackColor);
+    final Paint activePaint = Paint()..color = activeTrackColorTween.evaluate(enableAnimation);
+    final Paint inactivePaint = Paint()..color = inactiveTrackColorTween.evaluate(enableAnimation);
+
+    final Rect trackRect = getPreferredRect(
+      parentBox: parentBox,
+      offset: offset,
+      sliderTheme: sliderTheme,
+      isEnabled: isEnabled,
+      isDiscrete: isDiscrete
+    );
+    final double thumbRadius = sliderTheme.rangeThumbShape.getPreferredSize(isEnabled, isDiscrete).width / 2.0;
+    final Rect leftTrackSegment = Rect.fromLTRB(trackRect.left, trackRect.top, startThumbCenter.dx - thumbRadius, trackRect.bottom);
+    context.canvas.drawRect(leftTrackSegment, inactivePaint);
+    final Rect middleTrackSegment = Rect.fromLTRB(startThumbCenter.dx + thumbRadius, trackRect.top, endThumbCenter.dx - thumbRadius, trackRect.bottom);
+    context.canvas.drawRect(middleTrackSegment, activePaint);
+    final Rect rightTrackSegment = Rect.fromLTRB(endThumbCenter.dx + thumbRadius, trackRect.top, trackRect.right, trackRect.bottom);
+    context.canvas.drawRect(rightTrackSegment, inactivePaint);
+  }
+}
+
+/// This is the default shape of a [RangeSlider]'s track.
+///
+/// It paints a solid colored rectangle with rounded edges, vertically centered
+/// in the [parentBox]. The track rectangle extends to the bounds of the
+/// [parentBox], but is padded by the larger of [RoundSliderOverlayShape]'s
+/// radius and [RoundRangeSliderThumbShape]'s radius. The height is defined by
+/// the [SliderThemeData.trackHeight]. The color is determined by the
+/// [RangeSlider]'s enabled state and the track segment's active state which are
+/// defined by:
+///   [SliderThemeData.activeTrackColor],
+///   [SliderThemeData.inactiveTrackColor],
+///   [SliderThemeData.disabledActiveTrackColor],
+///   [SliderThemeData.disabledInactiveTrackColor].
+///
+/// See also:
+///
+///  * [RangeSlider], for the component that is meant to display this shape.
+///  * [SliderThemeData], where an instance of this class is set to inform the
+///    slider of the visual details of the its track.
+///  * [RangeSliderTrackShape], which is the base component for creating other
+///    custom track shapes.
+///  * [RectangularRangeSliderTrackShape], for a similar track with sharp edges.
+class RoundedRectRangeSliderTrackShape extends RangeSliderTrackShape {
+  /// Abstract const constructor. This constructor enables subclasses to provide
```

Comment:
Looks like incorrect documentation for this one too, not an abstract class.

Reply:
Fixed